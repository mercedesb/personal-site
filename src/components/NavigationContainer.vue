<template>
  <div class="NavigationContainer">
    <Navigation :navLinks="navLinks" :homeLink="homeLink" />
    <MobileNavigation :navLinks="navLinks" :color="color" />
  </div>
</template>

<script>
import Navigation from './Navigation.vue'
import MobileNavigation from './MobileNavigation.vue'
import images from '../mixins/images'

export default {
  components: {
    Navigation, MobileNavigation
  },
  mixins: [
    images
  ],
  props: {
    color: String
  },
  computed: {
    navLinks () {
      return this.$store.state.navLinks.slice(1).map((link) => {
        return {
          parsedLink: link.externalLink ? link.externalLink : `/${link.urlSegment}`,
          ...link,
          iconSvg: link.iconSvg.fields.svg
        }
      })
    },
    homeLink () {
      const link = this.$store.state.navLinks[0]
      if (link) {
        return {
          parsedLink: '/',
          ...link,
          iconSvg: link.icon.fields.svg
        }
      } else {
        return {
          parsedLink: '/',
          iconSvg: ''
        }
      }
    }
  }
}
</script>

<style lang="scss">
  .NavigationContainer {
    width: 100%;
  }
</style>