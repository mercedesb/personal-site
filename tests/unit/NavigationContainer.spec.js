import NavigationContainer from "@/components/NavigationContainer.vue";
import { TestUtility } from "./TestUtility";

describe("NavigationContainer", () => {
  let component;

  const homePageNavLink = {
    icon: {
      fields: {
        svg: "<svg></svg>"
      }
    }
  };

  const landingPageNavLinks = Object.keys(TestUtility.landingPages).map(key => {
    return TestUtility.landingPages[key];
  });

  const state = {
    navLinks: [homePageNavLink, ...landingPageNavLinks]
  };

  const actions = {
    getNavLinks: jest.fn()
  };

  const store = {
    state,
    actions
  };

  const shallow = propsData =>
    TestUtility.shallow(NavigationContainer, {
      store,
      propsData: {
        ...propsData
      }
    });

  describe("Snapshots", () => {
    it("matches snapshot", () => {
      component = shallow();
      expect(component.element).toMatchSnapshot();
    });
  });

  describe("Computed", () => {
    describe("navLinks", () => {
      beforeEach(() => {
        component = shallow();
      });

      it("has navLinks populated", () => {
        expect(component.vm.navLinks).toBeDefined();
      });

      it("creates a parsedLink prop on the navLinks", () => {
        expect(component.vm.navLinks[0].parsedLink).toBeDefined();
      });

      it("uses the external link for parsedLink if its defined", () => {
        const shopPage = TestUtility.landingPages["shop"];
        const shopLink = component.vm.navLinks.find(link => {
          return link.title === shopPage.title;
        });
        expect(shopLink.parsedLink).toEqual(shopPage.externalLink);
      });

      it("uses the url segment for parsedLink if eternalLink is not defined", () => {
        const blogPage = TestUtility.landingPages["blog"];
        const blogLink = component.vm.navLinks.find(link => {
          return link.title === blogPage.title;
        });
        expect(blogLink.parsedLink).toEqual(`/${blogPage.urlSegment}`);
      });
    });
  });
});
