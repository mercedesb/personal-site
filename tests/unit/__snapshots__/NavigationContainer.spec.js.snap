// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationContainer Snapshots matches snapshot 1`] = `
<navigation-stub
  homelink="[object Object]"
  navlinks="[object Object],[object Object],[object Object],[object Object]"
/>
`;
